### PR TITLE
style(js): prefer const for non-reassigned values

### PR DESCRIPTION
**Pull request by mcous**
## overview

This PR makes a couple lint config changes to make things a little easier on our brains:

- Disallow `var`
    - We had `var`'s in our codebase!
- Prefer `const` over `let` if variable isn't reassigned

## changelog

All code changes (except for one which needed manual intervention) were made with `eslint --fix`

- style(js): prefer const for non-reassigned values

## review requests

- [ ] Changes look good

## risk assessment

Medium-low. Can't think of any logical reason why this would be dangerous but the PR touches a wide variety of files

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -17,11 +17,8 @@ module.exports = {
 
   rules: {
     camelcase: 'off',
-    // TODO(mc, 2019-06-28): these flowtype rules are noisy (~1000 warnings),
```
Comment (mcous) [inline .eslintrc.js:20]: This TODO was no longer valid; the referenced rules don't really work

File: app/src/rpc/client.js [hunk 2]
```diff
@@ -72,7 +72,10 @@ class RpcContext extends EventEmitter {
     const failureEvent = makeFailureEventName(token)
 
     return new Promise((resolve, reject) => {
-      let timeout
+      const timeout = setTimeout(
```
Comment (mcous) [inline app/src/rpc/client.js:75]: This was the manual intervention

**Review by IanLondon — APPROVED** [review]
code changes look good, seems safe if CI passes. As much as I wish `const` that indicated an object or array was intended to be immutable and `let` indicated that it is intended to be mutable, that's not how JS actually works! I like this change.

**Review by shlokamin — APPROVED** [review]
Good stuff, thanks for this!

**Pull request merged**